src/schema/users.ts

```diff
@@ -694,6 +704,36 @@ export const resolvers: IResolvers<any, Context> = {
       ctx: Context,
       info,
     ): Promise<Connection<GQLView>> => readHistoryResolver(args, ctx, info),
+    generateUniqueUsername: async (
+      _,
+      args: { name: string },
+      ctx: Context,
+    ): Promise<string> => {
+      const name = args.name.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
+      let attemptCount = 0;
+
+      if (name.length === 0) {
+        return '';
+      }
+
+      let username = name.substring(0, 39);
+
+      while (attemptCount < 3) {
```

Comment:
I think to improve the performance check, we can generate 3 usernames already, then do a single `findBy` query so we won't have to do 3 awaits worst-case scenario. Then decide from there which to send. We can probably return an array of strings to provide multiple suggestions but probably out of scope for now.

Reply:
That's a great suggestion, I love it! We can even create more than 3 if we do it in one query